pkg/collector/python/datadog_agent.go

```diff
@@ -609,3 +612,70 @@ func ObfuscateMongoDBString(cmd *C.char, errResult **C.char) *C.char {
 	// memory will be freed by caller
 	return TrackedCString(obfuscatedMongoDBString)
 }
+
+var (
+	telemetryMap = new(sync.Map)
+)
+
+func lazyInitTelemetryHistogram(checkName string, metricName string) telemetry.Histogram {
+	var key = checkName + "." + metricName
+	histogram, _ := telemetryMap.LoadOrStore(key, telemetryimpl.GetCompatComponent().NewHistogramWithOpts(
```

Comment:
I am pretty sure this is going to panic when called the second time for the same metric.

Reply:
Refactored to use a manual lock to avoid registering metrics twice, and updated tests to check this better.